### PR TITLE
Add gradient tests

### PR DESCRIPTION
**Pull request by gomezzz**
# Description
- Fixed gradient flow in all integration methods
- Slight refactoring of VEGAS to fix gradient flow
- Small fix in integration_test_utils
- Removed some unused imports
- Renamed a function to match naming conventions
- Added gradient test (currently failing for VEGAS)
- Added function to create a linspace without breaking gradients

## Resolved Issues

Add gradient tests

## How Has This Been Tested?

- [x] Unit tests
- [x] New gradient test

## Related Pull Requests
N/A

### CODE REVIEW

File: torchquad/tests/gradient_test.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    N = 99997
+    result_tolerence = 1e-2
+    gradient_tolerance = 1e-2
+    torch.manual_seed(0)  # we have to seed torch to get reproducible results
+
+    # Define integrators
+    integrators = [MonteCarlo(), Trapezoid(), Simpson(), Boole(), VEGAS()]
+    for integrator in integrators:
+        # Compute integral
```
Comment (GabrieleMeoni) [inline torchquad/tests/gradient_test.py:39]: Can you print what method are you testing? Otherwise, it is difficult to get what method is failing in case of failure.

**Review by GabrieleMeoni — CHANGES_REQUESTED** [review]

File: torchquad/integration/utils.py [hunk 2]
```diff
@@ -6,7 +6,34 @@
 logger = logging.getLogger(__name__)
 
 
-def setup_integration_domain(dim, integration_domain):
+def _linspace_with_grads(start, stop, N, requires_grad):
+    """Creates a equally spaced 1D grid while keeping gradients
```
[the remark this answers is not in the document]
Comment (gomezzz) [reply torchquad/integration/utils.py:10]:
```suggestion
    """Creates an equally spaced 1D grid while keeping gradients
```

File: torchquad/tests/gradient_test.py (hunk 1 above)
Comment (gomezzz) [reply torchquad/tests/gradient_test.py:41]: Done

File: torchquad/integration/integration_grid.py [hunk 3]
```diff
@@ -41,12 +43,22 @@ def __init__(self, N, integration_domain):
             + " points over"
             + str(integration_domain),
         )
+
+        # Check if domain requires gradient
```
Comment (gomezzz) [inline torchquad/integration/integration_grid.py:47]:
```suggestion
        # Check if domain requires gradient here
```

**Review by GabrieleMeoni — APPROVED** [review]
